h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -134,8 +136,8 @@ public Frame(Key<Frame> key, String names[], Vec vecs[] ) {
   }
 
   public void setNames(String[] columns){
-    if(columns.length!= _vecs.length){
-      throw new IllegalArgumentException("Size of array containing column names does not correspond to the number of vecs!");
+    if (_vecs != null && columns.length != _vecs.length) {
+      throw new IllegalArgumentException("Number of column names=" + columns.length + " must be the number of vecs=" + _vecs.length);
```

Comment:
throw new IllegalArgumentException("Number of column names=" + columns.length + " must equal to the number of vecs=" + _vecs.length);